tests/repository_test.py

```diff
@@ -279,14 +280,123 @@ def test_node_hook_with_npm_userconfig_set(tempdir_factory, store, tmpdir):
+        *(expected_hook_expr or (expected_path,)),
+        *expected_args,
+    )
+    assert ret == expected
+
+
+def test_renv_parsing_file_no_opts_no_args(tempdir_factory, store):
+    hook_id = 'parse-file-no-opts-no-args'
+    _test_renv_parsing(tempdir_factory, store, hook_id)
+
+
+def test_renv_parsing_file_opts_no_args(tempdir_factory, store):
+
+    with pytest.raises(SyntaxError, match='only valid'):
+        renv.entry_validate(['Rscript', '--no-init', '/path/to/file'])
+
+
+def test_renv_parsing_file_no_opts_args(tempdir_factory, store):
+    hook_id = 'parse-file-no-opts-args'
+    expected_args = ['--no-cache']
+    _test_renv_parsing(
+        tempdir_factory, store, hook_id, expected_args=expected_args,
+    )
+
+
+def test_renv_parsing_expr_no_opts_no_args1(tempdir_factory, store):
+    hook_id = 'parse-expr-no-opts-no-args-1'
+    _test_renv_parsing(
+        tempdir_factory, store, hook_id, expected_hook_expr=('-e', '1+1'),
+    )
+
+
+def test_renv_parsing_expr_no_opts_no_args2(tempdir_factory, store):
+    with pytest.raises(SyntaxError, match='at most one'):
+        renv.entry_validate(['Rscript', '-e', '1+1', '-e', 'letters'])
+
+
+def test_renv_parsing_expr_opts_no_args2(tempdir_factory, store):
+    with pytest.raises(SyntaxError, match='only valid'):
+        renv.entry_validate(
+            [
+                'Rscript', '--vanilla', '-e', '1+1', '-e', 'letters',
+            ],
+        )
+
+
+def test_renv_parsing_expr_args_in_entry2(tempdir_factory, store):
+    with pytest.raises(SyntaxError, match='at most one'):
+        renv.entry_validate(['Rscript', '-e', 'expr1', '--another-arg'])
+
+
+def test_renv_parsing_expr_non_Rscirpt(tempdir_factory, store):
+
+    with pytest.raises(SyntaxError, match='entry must start with'):
+        renv.entry_validate(['AnotherScript', '-e', '{{}}'])
+
+
+def test_renv_hook(tempdir_factory, store):
+    _test_hook_repo(
+        tempdir_factory, store, 'renv_hooks_repo',
+        'hello-world', [os.devnull],
+        b'Hello, World, from R!\n',
+    )
+
+
+def test_renv_inline_hook(tempdir_factory, store):
+    _test_hook_repo(
+        tempdir_factory, store, 'renv_hooks_repo',
+        'hello-world-inline', ['some-file'],
+        b'Hi-there, some-file, from R!\n',
+    )
+
+
+def test_renv_with_additional_dependencies_hook(tempdir_factory, store):
+    _test_hook_repo(
+        tempdir_factory, store, 'renv_hooks_repo',
+        'additional-deps', [os.devnull],
+        b'OK\n',
+        config_kwargs={
+            'hooks': [{
+                'id': 'additional-deps',
+                'additional_dependencies': ['cachem@1.0.4'],
+            }],
+        },
+    )
+
+
 def test_run_a_ruby_hook(tempdir_factory, store):
     _test_hook_repo(
         tempdir_factory, store, 'ruby_hooks_repo',
         'ruby_hook', [os.devnull], b'Hello world from a ruby hook\n',
     )
 
 
-@xfailif_windows  # pragma: win32 no cover
+@ xfailif_windows  # pragma: win32 no cover
```

Comment:
decorators shouldn't have spaces after them -- not sure what happened here

Reply:
Also wondered, but one of the pre-commit hook changed it, maybe autopep? I even rebased my PR master and tried to get rid of this. Can revert but then not sure if the pre-commit.ci passes.

Reply:
ah my guess is there was a syntax error earlier in the file and then autopep8 freaked -- I've been meaning to fix that in autopep8 but it's tricky (basically sometimes it will try and rewrite a file even if it has a syntax error 😱)